HPCC-25955 implemented static pvs

rrao4 opened the pull request:
Implemented static pvs in order to create storage beyond the eks cluster.
Signed off by raor0000@risk.regn.net
<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->
I tested the changes using AWS EKS.
<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

Commit:
HPCC-29381 Improve XSD generation for ESP services

Replaced explicit XSD and WSDL generation in genreated code by
using existing library function.

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

ghalliday committed:
Split off 9.2.0

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

# Conflicts:
#	helm/hpcc/Chart.yaml
#	helm/hpcc/templates/_helpers.tpl
#	helm/hpcc/templates/dafilesrv.yaml
#	helm/hpcc/templates/dali.yaml
#	helm/hpcc/templates/dfuserver.yaml
#	helm/hpcc/templates/eclagent.yaml
#	helm/hpcc/templates/eclccserver.yaml
#	helm/hpcc/templates/eclscheduler.yaml
#	helm/hpcc/templates/esp.yaml
#	helm/hpcc/templates/localroxie.yaml
#	helm/hpcc/templates/roxie.yaml
#	helm/hpcc/templates/sasha.yaml
#	helm/hpcc/templates/thor.yaml
#	version.cmake

ghalliday committed:
HPCC-29858 Update the platform version rules

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

jakesmith committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

Commit:
HPCC-29744 Remove existing 'optional' keywords from SCM files

Removed 'optional' keywords from existing SCM files because they
were not added correctly. The keyword requires a string value in
the form optional("value") where "value" creates a category of
optional fields exposed via the interface when the category
"value" is included as a URL query parameter.

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

ghalliday committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

richardkchapman committed:
HPCC-29831 Allow localAgent (and other) setting to be set in workunit/regression suite

These options are only applied when Roxie us run in workunit mode in containerized systems.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

GordonSmith committed:
HPCC-29828 Dev docs failing to build/publish

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

richardkchapman committed:
HPCC-29838 Add false sharing test case

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #17512 from GordonSmith/HPCC-29828-DEV_DOCS

HPCC-29828 Dev docs failing to build/publish

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>
Merged-by: Gavin Halliday <ghalliday@hpccsystems.com>

GordonSmith committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

GordonSmith committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

GordonSmith committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

ghalliday committed:
Merge pull request #17505 from kenrowland/HPCC-29744

HPCC-29744 Remove existing 'optional' keywords from SCM files

Reviewed-By: Shamser Ahmed <shamser.ahmed@lexisnexis.com>
Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexisrisk.com>
Merged-by: Gavin Halliday <ghalliday@hpccsystems.com>

jakesmith committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

GordonSmith committed:
Merge pull request #17534 from ghalliday/versiondetails

HPCC-29858 Update the platform version rules

Commit:
Add dependency so unit tests will build

GordonSmith committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

# Conflicts:
#	helm/hpcc/Chart.yaml
#	helm/hpcc/templates/_helpers.tpl
#	helm/hpcc/templates/dafilesrv.yaml
#	helm/hpcc/templates/dali.yaml
#	helm/hpcc/templates/dfuserver.yaml
#	helm/hpcc/templates/eclagent.yaml
#	helm/hpcc/templates/eclccserver.yaml
#	helm/hpcc/templates/eclscheduler.yaml
#	helm/hpcc/templates/esp.yaml
#	helm/hpcc/templates/localroxie.yaml
#	helm/hpcc/templates/roxie.yaml
#	helm/hpcc/templates/sasha.yaml
#	helm/hpcc/templates/thor.yaml
#	version.cmake

GordonSmith committed:
Merge pull request #17349 from kenrowland/HPCC-29381

HPCC-29381 Improve XSD generation for ESP services

rrao4 committed:
HPCC-25955 implemented static pvs

xwang2713 commented:
@g-pan please remove examples/efs/README.md

GordonSmith committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

# Conflicts:
#	helm/hpcc/Chart.yaml
#	helm/hpcc/templates/_helpers.tpl
#	helm/hpcc/templates/dafilesrv.yaml
#	helm/hpcc/templates/dali.yaml
#	helm/hpcc/templates/dfuserver.yaml
#	helm/hpcc/templates/eclagent.yaml
#	helm/hpcc/templates/eclccserver.yaml
#	helm/hpcc/templates/eclscheduler.yaml
#	helm/hpcc/templates/esp.yaml
#	helm/hpcc/templates/localroxie.yaml
#	helm/hpcc/templates/roxie.yaml
#	helm/hpcc/templates/sasha.yaml
#	helm/hpcc/templates/thor.yaml
#	version.cmake

Commit:
HPCC-29963 Incorrect XSD type output for TpQueryType in WsTopology service

Changed TpQueryType to string to produce correct type in XSD

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

wangkx committed:
HPCC-29972 Add WsTopology.TpListLogFiles

The existing code uses the FileSpray.FileList to list the log files
of each HPCC component in bare metal enviroment, which does not make
sense. In this PR, a similar funcitonality is added to WsTopology.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

g-pan commented on helm/examples/efs/README.md:
````diff
@@ -51,45 +52,114 @@ efs-csi-controller-594c7f67c7-mncgd   3/3     Running   0          32h
+  helm uninstall myhpcc
+```
+## 2. Static storage within Kubernetes with values-retained-efs.yaml
+In this method, storage lives on the Kubernetes cluster level. It uses the helm chart "hpcc-efs-dynamic-pv" to manually create PVCs. The PVs are dynamically generated. The PVCs will persist after the HPCC cluster is deleted, meaning that the storage can be reused across different HPCC clusters.<br/>
````
The PVs are dynamically,  should it be PVCs ??

rrao4 replied:
No, the PVCs aren't dynamically created, in the hpcc-efs-dynamic-pv/templates/create-pvc.yaml, they are actually manually created.

g-pan reviewed (commented):
Only one comment/question otherwise approved.

xwang2713 commented on helm/examples/efs/create-ap.sh:
Add comment that For RISK user tag "owner" and "owner_email" are may required

rrao4 replied:
I added this just now to the README under the explanation of the 3rd storage method.

xwang2713 reviewed (commented):
Generally looks good. The tags need "owner" and "owner_email" for RISK users.

GordonSmith committed:
Merge pull request #17515 from richardkchapman/false-sharing

HPCC-29838 Add false sharing test case

GordonSmith committed:
Merge pull request #17612 from kenrowland/HPCC-29963

HPCC-29963 Correct XSD type conversion for TpQueryType in WsTopology Service

GordonSmith committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

# Conflicts:
#	helm/hpcc/Chart.yaml
#	helm/hpcc/templates/_helpers.tpl
#	helm/hpcc/templates/dafilesrv.yaml
#	helm/hpcc/templates/dali.yaml
#	helm/hpcc/templates/dfuserver.yaml
#	helm/hpcc/templates/eclagent.yaml
#	helm/hpcc/templates/eclccserver.yaml
#	helm/hpcc/templates/eclscheduler.yaml
#	helm/hpcc/templates/esp.yaml
#	helm/hpcc/templates/localroxie.yaml
#	helm/hpcc/templates/roxie.yaml
#	helm/hpcc/templates/sasha.yaml
#	helm/hpcc/templates/thor.yaml
#	version.cmake

richardkchapman committed:
HPCC-30005 Refactor IMessageUnpackCursor

The way record lengths were fetched caused significant extra work for the localAgent
where a temporary roxie row would be allocated and released for each count.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
HPCC-29831 Allow localAgent (and other) setting to be set in workunit/regression suite

Fix some issues with setting topology when localAgent is false

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-27303 Coverity scan reported new defects related to Configuration

Changed loop to return immediately when element erased from map

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

jakesmith committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

# Conflicts:
#	helm/hpcc/Chart.yaml
#	helm/hpcc/templates/_helpers.tpl
#	helm/hpcc/templates/dafilesrv.yaml
#	helm/hpcc/templates/dali.yaml
#	helm/hpcc/templates/dfuserver.yaml
#	helm/hpcc/templates/eclagent.yaml
#	helm/hpcc/templates/eclccserver.yaml
#	helm/hpcc/templates/eclscheduler.yaml
#	helm/hpcc/templates/esp.yaml
#	helm/hpcc/templates/localroxie.yaml
#	helm/hpcc/templates/roxie.yaml
#	helm/hpcc/templates/sasha.yaml
#	helm/hpcc/templates/thor.yaml
#	version.cmake

g-pan reviewed (commented):
couple of comments in line.

rrao4 commented on helm/examples/efs/README.md:
```diff
@@ -125,3 +196,9 @@ To display the mount targets:
+Navigate to the EFS service and click "Create file system". Select the same VPC as the one your EKS cluster is running in.
+### Configuring the Mount Targets
+Click on the file system you just created and navigate to "Network". Your mount targets should be displayed.<br/>
+Click on "Manage", and you should see that the security groups for the mount targets are the default security group. Replace these with the EKS cluster security group. To find the EKS cluster security group, navigate to EKS, click on your cluster, click on "Networking", and it is the security group displayed under "Cluster security group". After replacing the default security group for all mount  ...
```
Thanks for the review. I've updated the parts you mentioned.

g-pan reviewed (approved):
approved.

ghalliday committed:
Merge pull request #17643 from kenrowland/HPCC-27303

HPCC-27303 Coverity scan reported new defects related to Configuration

Reviewed-By: Attila Vamos <attila.vamos@gmail.com>
Merged-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #17622 from wangkx/h29972

HPCC-29972 Add WsTopology.TpListLogFiles

Reviewed-By: Anthony Fishbeck <anthony.fishbeck@lexisnexisrisk.com>
Merged-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #17640 from richardkchapman/messageunpacker

HPCC-30005 Refactor IMessageUnpackCursor

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>
Merged-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #17513 from richardkchapman/localagent-wu

HPCC-29831 Allow localAgent setting to be set in workunit/regression suite

Reviewed-by: Jake Smith <jake.smith@lexisnexisrisk.com>
Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>
Merged-by: Gavin Halliday <ghalliday@hpccsystems.com>

xwang2713 reviewed (approved):
Ready to merge

ghalliday commented:
Please squash ready for merging

jakesmith committed:
Merge remote-tracking branch 'origin/candidate-9.2.x'

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

# Conflicts:
#	helm/hpcc/Chart.yaml
#	helm/hpcc/templates/_helpers.tpl
#	helm/hpcc/templates/dafilesrv.yaml
#	helm/hpcc/templates/dali.yaml
#	helm/hpcc/templates/dfuserver.yaml
#	helm/hpcc/templates/eclagent.yaml
#	helm/hpcc/templates/eclccserver.yaml
#	helm/hpcc/templates/eclscheduler.yaml
#	helm/hpcc/templates/esp.yaml
#	helm/hpcc/templates/localroxie.yaml
#	helm/hpcc/templates/roxie.yaml
#	helm/hpcc/templates/sasha.yaml
#	helm/hpcc/templates/thor.yaml
#	version.cmake

rrao4 commented:
Hi Gavin,

I rebased the commits into one, which is "HPCC-25955 implemented static pvs". Let me know if its ok to merge, or if I should create a new pull request.

Thanks,
Ryan Rao

ghalliday commented:
I think you need to execute something like:

git fetch upstream
git rebase --onto upstream/candidate-9.2.x HEAD~1

Assuming upstream refers to hpcc-systems/HPCC-Platform

To remove all the other commits from your PR.  I suspect your local candidate-9.2.x branch is out of sync with the hpcc-systems branch.

rrao4 commented:
Closed, new PR #17687 has been created with the correct base and same changes.
